pip/utils/__init__.py

```diff
@@ -622,7 +623,7 @@ def remove_tracebacks(output):
 def call_subprocess(cmd, show_stdout=True, cwd=None,
                     raise_on_returncode=True,
                     command_level=std_logging.DEBUG, command_desc=None,
-                    extra_environ=None, spinner=None):
+                    extra_environ=None, spinner=None, warn_on_returncode=True):
```

Comment:
Instead of 2 argumrents `raise_on_returncode` and `warn_on_returncode`, it might make more sense to have a single `on_returncode` in ('raise', 'warn', 'ignore').